specs/core/1_shard-data-chains.md

```diff
@@ -247,36 +250,37 @@ def verify_shard_attestation_signature(state: BeaconState,
     )
 ```
 
+### `flatten_block`
+
+```python
+def flatten_block(block: ShardBlock) -> Bytes:
+    return zpad(serialize(get_shard_header(block)), BYTES_PER_SHARD_BLOCK_HEADER) + serialize(block.body)
+```
+
+### `compute_crosslink_data`
+
+```python
+def compute_crosslink_data(blocks: Sequence[ShardBlock]) -> bytes:
+    """
+    Flattens a series of blocks. Designed to be equivalent to SSZ serializing a list of
+    flattened blocks with one empty block at the end as a termination marker.
+    """
+    positions = [4 * len(blocks)]
+    bodies = []
+    for block in blocks:
+        bodies.append(flatten_block(block))
```

Comment:
Doesn't flattening the block here reduce our ability to construct proofs about components of a block through the crosslink root? Still possible but not as clean due to flattening to bytes